django/db/models/fields/files.py

```diff
@@ -441,7 +441,8 @@ def contribute_to_class(self, cls, name, **kwargs):
         # after their corresponding image field don't stay cleared by
         # Model.__init__, see bug #11196.
         # Only run post-initialization dimension update on non-abstract models
-        if not cls._meta.abstract:
+        # with width_field/height_field.
+        if not cls._meta.abstract and (self.width_field or self.height_field):
```

Comment:
We should test this change, maybe:
```diff
diff --git a/tests/model_fields/test_imagefield.py b/tests/model_fields/test_imagefield.py
index 6ef46ef200..ca7ad2867e 100644
--- a/tests/model_fields/test_imagefield.py
+++ b/tests/model_fields/test_imagefield.py
@@ -5,6 +5,7 @@ from unittest import skipIf
 from django.core.exceptions import ImproperlyConfigured
 from django.core.files import File
 from django.core.files.images import ImageFile
+from django.db.models import signals
 from django.test import TestCase
 from django.test.testcases import SerializeMixin
 
@@ -328,6 +329,15 @@ class ImageFieldNoDimensionsTests(ImageFieldTwoDimensionsTests):
 
     PersonModel = Person
 
+    def test_post_init_not_connected(self):
+        person_model_id = id(PersonModel)
+        self.assertFalse(
+            any(
+                sender_id == person_model_id
+                for (_, sender_id), *_ in signals.post_init.receivers
+            )
+        )
+
 
 @skipIf(Image is None, "Pillow is required to test ImageField")
 class ImageFieldOneDimensionTests(ImageFieldTwoDimensionsTests):

```
Not sure if is there a better way :thinking:

Reply:
Alternatively:

```python
self.assertNotIn(
    person_model_id,
    [sender_id for (_, sender_id), *_ in signals.post_init.receivers]
)
```